Sources/Components/SelectionBoxes/SelectionIndicatorView.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+
+public class SelectionIndicatorView: UIView {
+    let imageView: AnimatedImageView = {
+        let view = AnimatedImageView(frame: .zero)
+        view.contentMode = .scaleAspectFit
+        view.animationRepeatCount = 1
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    public var isSelected: Bool = false {
+        didSet {
+            animateImage(selected: isSelected)
+        }
+    }
+
+    public var unselectedImage: UIImage? {
+        didSet {
+            imageView.image = unselectedImage
+        }
+    }
+
+    public var unselectedAnimationImages: [UIImage]? {
+        didSet {
+            imageView.animationImages = unselectedAnimationImages
+            if let unselected = unselectedAnimationImages {
+                imageView.unselectedDuration = Double(unselected.count) / AnimatedImageView.framesPerSecond
+            }
+        }
+    }
+
+    public var selectedImage: UIImage? {
+        didSet {
+            imageView.highlightedImage = selectedImage
+        }
+    }
+
+    public var selectedAnimationImages: [UIImage]? {
+        didSet {
+            imageView.highlightedAnimationImages = selectedAnimationImages
+            if let selected = selectedAnimationImages {
+                imageView.selectedDuration = Double(selected.count) / AnimatedImageView.framesPerSecond
+            }
+        }
+    }
+
+    public var type: SelectionBoxType = .check
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupStyle()
+        setupSubviews()
+    }
+
+    init(type: SelectionBoxType) {
+        self.type = type
+        super.init(frame: .zero)
+        setupStyle()
+        setupSubviews()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupSubviews() {
+        translatesAutoresizingMaskIntoConstraints = false
+        addSubview(imageView)
+        imageView.fillInSuperview()
+    }
+
+    private func setupStyle() {
+        let framesPerSecond = 60.0
+
+        let radiobuttonSelected = UIImage.animatedImageNamed(type.selected, duration: 13 / framesPerSecond)
```

Comment:
FYI, this wont work when the framework is used since `animatedImageNamed` uses the main bundle.

Reply:
mmmm you say it doesn't work but it works when I tried... so what do you mean?

Reply:
It does not work from the FINN app because it uses the FinniverseKit as a Framework. But the Demo app is using the code directly.

Reply:
If you try to report an ad from the FINN app it seems to be working. 🤔

Reply:
I found out why it works in the ReporterView. It manually gets all the images instead of using `UIImage.animatedImageNamed`
```
for i in 0 ..< 13 {
    if let image = UIImage(named: "radiobutton-select-\(i)", in: FinniversKit.bundle, compatibleWith: nil) {
        radioButtonSelected.append(image)
    }
}

for i in 0 ..< 10 {
    if let image = UIImage(named: "radiobutton-unselected-\(i)", in: FinniversKit.bundle, compatibleWith: nil) {
        radioButtonUnselected.append(image)
    }
}
```

Reply:
Ahhhh sneaky!